features/steps/definitions.py

```diff
@@ -99,14 +100,22 @@ def create_security_group_rule(context, security_group_name: str, direction: str
 
     @then("I should be able to delete routers")
     def delete_a_router(context):
-        for router in context.client.network.routers():
-            if context.test_name in router.name:
-                assert router is not None, f"Router with name {router.name} doesn't exist"
-                context.client.network.delete_router(router)
+        if context.collector.routers:
+            for router_id in context.collector.routers:
+                context.client.network.delete_router(router_id)
                 time.sleep(2)
                 assert not context.client.network.find_router(
-                    name_or_id=router.name
-                ), f"Router with name {router.name} was not deleted"
+                        name_or_id=router_id
+                ), f"Router with ID {router_id} was not deleted"
+        else:
```

Comment:
create new function verify_deleted router(may be done in tools) to check if all of the routers from test are deleted

Reply:
check also if len(context.collector.routers) == 0

Reply:
done